packages/remix-server-runtime/server.ts

```diff
@@ -176,14 +180,14 @@ async function handleDocumentRequest({
   loadContext,
   matches,
   request,
-  routes,
+  rootRoute,
   serverMode,
 }: {
   build: ServerBuild;
   loadContext?: AppLoadContext;
   matches: RouteMatch<ServerRoute>[] | null;
   request: Request;
-  routes: ServerRoute[];
+  rootRoute?: ServerRoute;
```

Comment:
We only ever use the root route inside here so I updated for clarity and to reduce the scope of the server-runtime usage of hierarchical `routes`